packages/wonder-stuff-i18n/src/plugins/i18n-plugin.js

```diff
@@ -0,0 +1,410 @@
+            getLocalePath: (locale) => `/${locale}/`,
+            shouldLocalizeAsset: (assetName) =>
+                assetName.startsWith("runtime."),
+            shouldProcessAsset: () => true,
+            shouldLocalizeManifest: (assetName) => assetName.endsWith(".html"),
+            ...options,
+            // Filter out English locale if it exists
+            locales: options.locales.filter((locale) => locale !== "en"),
+        };
+
+        if (this.options.locales.length === 0) {
+            this.log("I18nPlugin: No locales to localize to.");
+        }
+    }
+
+    /* istanbul ignore next */
+    apply(compiler: $FlowFixMe) {
+        compiler.hooks.emit.tapAsync("I18nPlugin", (options, callback) =>
+            this.onEmit(options, callback),
+        );
+    }
+
+    /**
+     * Log out something to the console, if we're not silent.
+     *
+     * @param  {...any} args the args to log out to the console
+     */
+    log(...args: Array<$FlowFixMe>) {
+        const {silent} = this.options;
+        if (silent === false || silent == null) {
+            // eslint-disable-next-line no-console
+            console.log(...args);
+        }
+    }
+
+    /**
+     * Get all the translated strings from Crowdin. The results are stored and
+     * returned in a map of locale to string map.
+     *
+     * @param {Array<string>} locales the locales to get the strings for
+     * @returns {Object<string, Object<string, Object>>} a mapping of locale to
+     *                                                   msgId to string data
+     */
+    async getTranslatedStrings(
+        locales: Array<string>,
+    ): Promise<TranslatedStrings> {
+        const {timing, getI18nStrings} = this.options;
+        const translatedStrings: TranslatedStrings = {};
+
+        const timingName = `I18nPlugin/Translation/get-strings`;
+
+        if (timing) {
+            timing.start(timingName);
+        }
+
+        // Download the strings for each locale in parallel
+        await asyncEachLimit(locales, 4, async (locale) => {
+            this.log(
+                `Retreiving strings for ${locale} ${getEmojiForLocale(locale)}`,
+            );
+
+            translatedStrings[locale] = await getI18nStrings(locale);
+        });
+
+        if (timing) {
+            timing.end(timingName);
+        }
+
+        return translatedStrings;
+    }
+
+    /**
+     * Go through all of the JS assets and insert translated strings for every
+     * locale that we're currently translating into.
+     *
+     * @param {assets: Object<string, Object>, translatedStrings: Object} options
+     * @returns {Object<string, string>} a map of old file hashes to new hashes
+     */
+    translateAssets({
+        assets,
+        translatedStrings,
+        localizeRuntimeFiles,
+        hashMap,
+    }: {
+        assets: Assets,
+        translatedStrings: TranslatedStrings,
+        localizeRuntimeFiles: boolean,
+        hashMap?: HashMaps,
+    }): HashMaps {
```

Comment:
Seeing all of these flow types in a webpack plugin fills my heart with immense joy. 😍